mindsdb/integrations/utilities/rag/loaders/vector_store_loader/vector_store_loader.py

```diff
@@ -0,0 +1,117 @@
+import ast
+import uuid
+
+from langchain_core.embeddings import Embeddings
+from langchain_community.vectorstores import Chroma, PGVector
+from langchain_core.vectorstores import VectorStore
+from pydantic import BaseModel
+
+from mindsdb.integrations.libs.vectordatabase_handler import TableField
+from mindsdb.integrations.utilities.rag.settings import VectorStoreType, VectorStoreConfig
+from mindsdb.utilities import log
+
+import pandas as pd
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import scoped_session, sessionmaker
+from sqlalchemy.exc import DisconnectionError
+
+logger = log.getLogger(__name__)
+
+
+class VectorStoreLoader(BaseModel):
+    embeddings_model: Embeddings
+    vector_store: VectorStore = None
+    config: dict = None
+
+    class Config:
+        arbitrary_types_allowed = True
+        extra = "forbid"
+        validate_assignment = True
+
+    def load(self) -> VectorStore:
+        """
+        Loads the vector store based on the provided config and embeddings model
+        :return:
+        """
+        self.vector_store = VectorStoreFactory.create(self.embeddings_model, self.config)
+        return self.vector_store
+
+
+class VectorStoreFactory:
+    @staticmethod
+    def create(embeddings_model: Embeddings, config: dict):
+
+        settings = VectorStoreConfig(**config)
+
+        if settings.vector_store_type == VectorStoreType.CHROMA:
+            return VectorStoreFactory._load_chromadb_store(embeddings_model, settings)
+        elif settings.vector_store_type == VectorStoreType.PGVECTOR:
+            return VectorStoreFactory._load_pgvector_store(embeddings_model, settings)
+        else:
+            raise ValueError(f"Invalid vector store type, must be one either {VectorStoreType.__members__.keys()}")
+
+    @staticmethod
+    def _load_chromadb_store(embeddings_model: Embeddings, settings) -> Chroma:
```

Comment:
this method (looking at the name) should be similar to `_load_pgvector_store`
why it doesn't use `_load_data_into_langchain_pgvector` ?

Reply:
loading an existing chromaDB instance works for all chromaDB even if created outside Langchain.

However, for PGVector if it is created outside of Langchain (i.e Mindsdb) we need to map data to langchain schema. see `_fetch_data_from_db`